packages/webapp/src/containers/CropCatalogue/useStringFilteredCrops.ts

```diff
@@ -49,8 +49,17 @@ export default function useStringFilteredCrops(
         'crop_cultivar', // crop_variety.crop_cultivar
       ];
       for (let i = 0; i < keys.length; i++) {
-        if (keys[i] in crop && check(crop[keys[i] as keyof typeof crop])) {
-          return true;
+        if (keys[i] in crop) {
+          if (!i) {
+            const scientificName =
+              crop[keys[0] as keyof typeof crop] + (crop[keys[1] as keyof typeof crop] || '');
+            if (scientificName && check(scientificName)) {
+              return true;
+            }
+            i++;
```

Comment:
Is this an efficiency to avoid an extra check during the loop since we know the next key is 'crop_variety_name'?

I am not sure I have seen for loop indexes bypassed this way before

Reply:
Thanks yah. The search function overall could be improved, we could probably use a library for this even to tokenize and allow for cross searching like "apple malus" working for apple. -- Future us problem haha

Reply:
Yeah, it could be an independent function at least. Tech debt candidate :)